UnitsNet/CustomCode/Quantities/Area.extra.cs

```diff
@@ -23,47 +23,5 @@ public static Area FromCircleRadius(Length radius)
         }
 
         #endregion
-
-        /// <summary>
-        /// Calculates the inverse of this quantity.
-        /// </summary>
-        /// <returns>The corresponding inverse quantity, <see cref="ReciprocalArea"/>.</returns>
-        public ReciprocalArea Inverse()
-        {
-            if (SquareMeters == 0.0)
-                return new ReciprocalArea(0.0, UnitsNet.Units.ReciprocalAreaUnit.InverseSquareMeter);
-
-            return new ReciprocalArea(1 / SquareMeters, UnitsNet.Units.ReciprocalAreaUnit.InverseSquareMeter);
-        }
```

Comment:
Are you confident that all existing operators still exist so we don't break anyone? 

Not so easy to verify in review, but if there is doubt we could maybe do some reflection at runtime before/after PR to list all quantities and their public methods, properties, operators and do a text diff on any difference in the public API.

Reply:
Yes, this is an interesting one. I'll see what I can do, but it's one of the reasons I would feel more comfortable with targeting v6 with this.

I know there are some longstanding architectural issues you would like to tackle too, but maybe those could be deferred to a v7? Then v6 could offer, amongst other things:

- Generated relational operators
- NaN, Inf values for double quantities
- Fixed typos in unit names
- Power using `double`

Reply:
Alright, using reflection to get the exact API changes was quite doable.

This PR adds 26 new operators to quantities and doesn't remove any.
```
AmountOfSubstance op_Multiply(System.TimeSpan, MolarFlow)              in MolarFlow
AmountOfSubstance op_Multiply(Duration, MolarFlow)                     in Duration
ElectricCharge op_Multiply(System.TimeSpan, ElectricCurrent)           in ElectricCurrent
ElectricCharge op_Multiply(ElectricCurrent, System.TimeSpan)           in ElectricCurrent
ElectricCurrent op_Division(ElectricCharge, System.TimeSpan)           in ElectricCharge
ElectricCurrent op_Multiply(System.TimeSpan, ElectricCurrentGradient)  in ElectricCurrentGradient
ElectricCurrent op_Multiply(Duration, ElectricCurrentGradient)         in Duration
Energy op_Multiply(Volume, EnergyDensity)                              in Volume
Force op_Multiply(System.TimeSpan, ForceChangeRate)                    in ForceChangeRate
Force op_Multiply(ForceChangeRate, System.TimeSpan)                    in ForceChangeRate
Force op_Multiply(Length, ForcePerLength)                              in Length
Jerk op_Division(Acceleration, System.TimeSpan)                        in Acceleration
LuminousIntensity op_Multiply(Area, Luminance)                         in Area
Mass op_Multiply(Area, AreaDensity)                                    in Area
Mass op_Multiply(Length, LinearDensity)                                in Length
MassFlow op_Multiply(MolarMass, MolarFlow)                             in MolarMass
Power op_Multiply(Area, HeatFlux)                                      in Area
Pressure op_Multiply(System.TimeSpan, PressureChangeRate)              in PressureChangeRate
Pressure op_Multiply(Duration, PressureChangeRate)                     in Duration
RotationalStiffness op_Multiply(Length, RotationalStiffnessPerLength)  in Length
Speed op_Multiply(System.TimeSpan, Acceleration)                       in Acceleration
Speed op_Multiply(Acceleration, System.TimeSpan)                       in Acceleration
Torque op_Multiply(Angle, RotationalStiffness)                         in Angle
Torque op_Multiply(Area, ForcePerLength)                               in Area
Volume op_Multiply(System.TimeSpan, VolumeFlow)                        in VolumeFlow
Volume op_Multiply(Mass, SpecificVolume)                               in Mass
```

29 existing operators get moved to a different quantity because we generate them as part of the left parameter:
```
Angle op_Multiply(Duration, RotationalSpeed)                   from RotationalSpeed to Duration
Area op_Multiply(Duration, KinematicViscosity)                 from KinematicViscosity to Duration
Energy op_Multiply(Duration, Power)                            from Power to Duration
Energy op_Multiply(Entropy, TemperatureDelta)                  from TemperatureDelta to Entropy
Energy op_Multiply(Mass, SpecificEnergy)                       from SpecificEnergy to Mass
Entropy op_Multiply(Mass, SpecificEntropy)                     from SpecificEntropy to Mass
Force op_Multiply(Acceleration, Mass)                          from Mass to Acceleration
Force op_Multiply(Area, Pressure)                              from Pressure to Area
ForcePerLength op_Multiply(Area, SpecificWeight)               from SpecificWeight to Area
Length op_Division(Force, ForcePerLength)                      from ForcePerLength to Force
Length op_Division(TemperatureDelta, TemperatureGradient)      from TemperatureGradient to TemperatureDelta
Length op_Multiply(Duration, Speed)                            from Speed to Duration
Mass op_Division(Mass, MassFraction)                           from MassFraction to Mass
Mass op_Multiply(Duration, MassFlow)                           from MassFlow to Duration
Mass op_Multiply(Mass, MassFraction)                           from MassFraction to Mass
Mass op_Multiply(MolarMass, AmountOfSubstance)                 from AmountOfSubstance to MolarMass
Mass op_Multiply(Volume, Density)                              from Density to Volume
Mass op_Multiply(Volume, MassConcentration)                    from MassConcentration to Volume
MassConcentration op_Multiply(Density, VolumeConcentration)    from VolumeConcentration to Density
MassConcentration op_Multiply(MolarMass, Molarity)             from Molarity to MolarMass
MassFlow op_Multiply(Density, VolumeFlow)                      from VolumeFlow to Density
Molarity op_Multiply(VolumeConcentration, Molarity)            from Molarity to VolumeConcentration
Power op_Multiply(Frequency, Energy)                           from Energy to Frequency
Power op_Multiply(Speed, Force)                                from Force to Speed
SpecificEnergy op_Multiply(SpecificEntropy, TemperatureDelta)  from TemperatureDelta to SpecificEntropy
TemperatureDelta op_Multiply(Duration, TemperatureChangeRate)  from TemperatureChangeRate to Duration
TemperatureDelta op_Multiply(Length, TemperatureGradient)      from TemperatureGradient to Length
Torque op_Multiply(Force, Length)                              from Length to Force
Volume op_Multiply(Area, Length)                               from Length to Area
```

Reply:
I missed this update, but that sounds great!
Will look over a bit more closely then I believe we are ready to merge.

Reply:
> 29 existing operators get moved to a different quantity

I think this should be safe and not break anyone, just a bit unsure if it's binary compatible. Will it break an app depending on `Library X`, which depends on UnitsNet, when the app upgrades UnitsNet and library still expects the old version.

Will look into it.

Reply:
Sorry this is taking so long, I'm just having a busy time with work not finding much time or energy for other things. But I intend to help wrap this up soon.

Reply:
Just to comment on versioning

> I know there are some longstanding architectural issues you would like to tackle too, but maybe those could be deferred to a v7

Generally speaking, I have over the years started to appreciate libs that make an effort to not break anyone when upgrading. It doesn't matter if we slap a major version bump on the change, it will still annoy the hell out of people if stuff stops working on upgrade or they are blocked by intermediate dependencies that also need to upgrade first. So the change better be worth it, trivial to fix or affect a very small group of users.

Reply:
I did a quick test on binary compatibility, and it seems moving the operator to a different class is a breaking change.

![image](https://github.com/angularsen/UnitsNet/assets/787816/ad93c1ec-c252-403b-bfe4-44f690f5a48e)


To reproduce:
1. Create local nuget of this PR's change (e.g. 5.99.0)
2. Add library Lib1, which depends on current latest 5.43.0, and performs multiplication on some given quantities

You get a `MissingMethodException`, because the multiply operator method no longer exists on `Power`, it was moved to `Duration`.

> Energy op_Multiply(Duration, Power)                            from Power to Duration

```cs
using System.Diagnostics;
using System.Reflection;
using Lib1;
using UnitsNet;

LogVersion(typeof(Lib1Helpers).Assembly);
LogVersion(typeof(Power).Assembly);

Energy energy1 = Lib1Helpers.Multiply(Power.FromWatts(100), Duration.FromSeconds(10));
Energy energy2 = Lib1Helpers.Multiply(Duration.FromSeconds(10), Power.FromWatts(100));

Console.WriteLine("Energy1: " + energy1);
Console.WriteLine("Energy2: " + energy2);
return;

void LogVersion(Assembly assembly)
{
    Console.WriteLine($"Assembly: {assembly.FullName}");
    Console.WriteLine("Version: " + assembly.GetName().Version);
    Console.WriteLine("File version: " + FileVersionInfo.GetVersionInfo(assembly.Location).FileVersion);
    Console.WriteLine("Product version: " + FileVersionInfo.GetVersionInfo(assembly.Location).ProductVersion);
    Console.WriteLine("\n---\n");
}
```

I don't think this will affect a lot of people, but it is definitely a breaking change, so let's target v6 branch. 
To refer back to my previous comment on breaking changes, this is one change I think is worth it because I don't think it will affect many users.

### In summary
- I'll retarget this PR to v6 and merge it there

Reply:
Cool! 👍 Thanks for your guidance with this.